pysteps/blending/steps.py

```diff
@@ -65,1723 +64,2858 @@
+                    fft_method=self.__params.fft_objs[j],
+                    domain=self.__config.domain,
+                )
+                epsilon_decomposed = self.__params.decomposition_method(
+                    epsilon,
+                    self.__params.bandpass_filter,
+                    fft_method=self.__params.fft_objs[j],
+                    input_domain=self.__config.domain,
+                    output_domain=self.__config.domain,
+                    compute_stats=True,
+                    normalize=True,
+                    compact_output=True,
+                )
+                self.__state.precip_mean_noise[j] = epsilon_decomposed["means"]
+                self.__state.precip_std_noise[j] = epsilon_decomposed["stds"]
+                for i in range(self.__config.n_cascade_levels):
+                    epsilon_temp = epsilon_decomposed["cascade_levels"][i]
+                    epsilon_temp *= self.__params.noise_std_coeffs[i]
+                    for n in range(self.__config.ar_order):
+                        self.__state.precip_noise_cascades[j][i][n] = epsilon_temp
+                epsilon_decomposed = None
+                epsilon_temp = None
+
+    def __determine_subtimesteps_and_nowcast_time_step(self, t, subtimestep_idx):
+        if self.__params.time_steps_is_list:
+            self.__state.subtimesteps = [
+                self.__params.original_timesteps[t_] for t_ in subtimestep_idx
+            ]
+        else:
+            self.__state.subtimesteps = [t]
 
-        extrap_kwargs["return_displacement"] = True
+        if (self.__params.time_steps_is_list and self.__state.subtimesteps) or (
+            not self.__params.time_steps_is_list and t > 0
+        ):
+            self.__state.is_nowcast_time_step = True
+        else:
+            self.__state.is_nowcast_time_step = False
 
-        forecast_prev = deepcopy(precip_cascade)
-        noise_prev = deepcopy(noise_cascade)
+        if self.__state.is_nowcast_time_step:
+            print(
+                f"Computing nowcast for time step {t}... ",
+                end="",
+                flush=True,
+            )
 
-        t_prev = [0.0 for j in range(n_ens_members)]
-        t_total = [0.0 for j in range(n_ens_members)]
+    def __decompose_nwp_if_needed_and_fill_nans_in_nwp(self, t):
+        if self.__state.precip_models_cascades is not None:
+            decomp_precip_models = list(self.__state.precip_models_cascades[:, t])
 
-        # iterate each time step
-        for t, subtimestep_idx in enumerate(timesteps):
-            if timestep_type == "list":
-                subtimesteps = [original_timesteps[t_] for t_ in subtimestep_idx]
+        else:
+            if self.__precip_models.shape[0] == 1:
+                decomp_precip_models = [
+                    self.__params.decomposition_method(
+                        self.__precip_models[0, t, :, :],
+                        bp_filter=self.__params.bandpass_filter,
```

Comment:
The problem is that this would mean we have to rewrite the decompositor importer code, and then we have breaking changes, so that is why I kept the left-hand side of these function arguments with the old names...

Reply:
Fair point, let's leave it like this then for now. This could be a TO DO for version 2, as breaking changes are okay then.

Reply:
Should I mark this as an issue or write a TODO line. It seems a bit to big of a tasks to put this as a one line TODO in one python file